File: src/main/java/net/frozenorb/potpvp/util/nametag/construct/NameTagInfo.java
package net.frozenorb.potpvp.util.nametag.construct;

import lombok.Getter;
import net.frozenorb.potpvp.util.packet.ScoreboardTeamPacketMod;

import java.util.ArrayList;

@Getter
public class NameTagInfo {

    private final String name;
    private final String prefix;
    private final String suffix;
    private final ScoreboardTeamPacketMod teamAddPacket;

    public NameTagInfo(String name, String prefix, String suffix) {
        this.name = name;
        this.prefix = prefix;
        this.suffix = suffix;

        teamAddPacket = new ScoreboardTeamPacketMod(name, prefix, suffix, new ArrayList<>(), 0);
    }
}

File: src/main/kotlin/net/frozenorb/potpvp/kt/nametag/NametagInfo.kt
package net.frozenorb.potpvp.kt.nametag

import net.frozenorb.potpvp.util.packet.ScoreboardTeamPacketMod
import java.util.ArrayList

class NametagInfo constructor(val name: String, val prefix: String, val suffix: String) {

    val teamAddPacket = ScoreboardTeamPacketMod(name, prefix, suffix, ArrayList(), 0)

    override fun equals(other: Any?): Boolean {
        if (other is NametagInfo) {
            val otherNametag = other as NametagInfo?
            return this.name == otherNametag!!.name && this.prefix == otherNametag.prefix && this.suffix == otherNametag.suffix
        }
        return false
    }

}
